### DISCUSSION TITLE
New shortcut action : "resume" and "resume match"

### DISCUSSION THREAD

**Discussion by MoonCactus** [jotaen/klog#257] [Feature ideas & discussions]
I wrapped klog around a dirty but pretty useful bash script, because I heavily rely on a "resume" action : it re-starts a task with the exact same tag and label of the last found in the journal.

Eg. I before I have a break, I first "stop". Then I simply "resume" when I'm back, no need to check and remember the task tag and description.

Actually I came here to ask whether it could be integrated officially because I recently upgraded it to something even more productive imho: I can "resume keyword". In this case I re-start the last task that matches (grep) the keyword.

It really makes my life easier when because I deal with 2 or 3 clients at the same time, so I keep switching from one to another (my bash script even had an "edit" option to open the journal and edit it by hand - for personal convenience I use only one journal, and one tag per client, I know it could be otherwise)

My two cents, hats off again for making this terrific tool !

**Comment by jotaen** [comment]
Yeah, I think “continuing” a previous activity via a new entry is a common use-case, and I actually also find myself running into this sometimes.

We already had discussions around this topic:

- https://github.com/jotaen/klog/discussions/237 about adding a CLI command that would start a new open time range and take over the previous entry summary. That’s pretty similar to what you have in mind, I think.
- https://github.com/jotaen/klog/discussions/239 is an idea how to address this on file format level, independent from the CLI tool.

In any event, since this has come up again now, I think it would be worth to add a flag for this: https://github.com/jotaen/klog/issues/258

**Reply by MoonCactus** [reply]
Indeed, you may better close this one. I added some comments on the linked entries
Sorry for this, I should have read better before posting :)

**Reply by jotaen** [reply]
No worries, it’s just a discussion thread (which can’t be closed anyway), so it can just stay as is 🙂

**Comment by jotaen** [comment]
`klog start --resume` will be included in the next release. (https://github.com/jotaen/klog/pull/259)
